collect_app/src/test/java/org/odk/collect/android/formentry/QuitFormDialogFragmentTest.java

```diff
@@ -0,0 +1,54 @@
+package org.odk.collect.android.formentry;
+
+import android.app.Dialog;
+
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.LooperMode;
+import org.robolectric.shadows.ShadowDialog;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+@LooperMode(LooperMode.Mode.PAUSED)
+public class QuitFormDialogFragmentTest {
+
+    private FragmentManager fragmentManager;
+    private QuitFormDialogFragment dialogFragment;
+
+    @Before
+    public void setup() {
+        FragmentActivity activity = Robolectric.setupActivity(FragmentActivity.class);
+        fragmentManager = activity.getSupportFragmentManager();
+        dialogFragment = new QuitFormDialogFragment();
+    }
+
+    @Test
+    public void dialogIsCancellable() {
+        dialogFragment.show(fragmentManager, "tag");
+
+        fragmentManager.executePendingTransactions();
```

Comment:
I think you can just do `shadowOf(getMainLooper()).idle();` instead here which a more general way of running the looper.

Reply:
I tried using `shadowOf(getMainLooper()).idle()`, but it doen't work. I stil get the error `main looper has queued unexecuted runnables` , unless I use `fragmentManager.executePendingTransactions()`.

Reply:
So you should be able to remove the `@LooperMode(PAUSED)` as well.

You'll see that you're getting that warning from Robolectric yes but the actual error is below that:

```
Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 5: TypedValue{t=0x2/d=0x7f040251 a=-1}
	at android.content.res.TypedArray.getColorStateList(TypedArray.java:546)
	at android.widget.TextView.readTextAppearance(TextView.java:3542)
	at android.widget.TextView.__constructor__(TextView.java:959)
	at android.widget.TextView.<init>(TextView.java)
	at android.view.LayoutInflater.createView(LayoutInflater.java:647)
	at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
	at android.view.LayoutInflater.onCreateView(LayoutInflater.java:720)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:788)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
	at org.odk.collect.android.adapters.IconMenuListAdapter.createView(IconMenuListAdapter.java:69)
	at org.odk.collect.android.adapters.IconMenuListAdapter.getView(IconMenuListAdapter.java:62)
```

This probably means that a layout is trying to grab an attribute (like `?colorSecondary` or `?attr/colorSecondary`) that isn't defined in the theme that's being used. It might be that something is using an attribute from our own theme or the `MaterialComponents` theme.

Reply:
Thanks @seadowg! I removed and got better results. I need not use `fragmentManager.executePendingTransaction` anymore.